### ISSUE TITLE
The trainer rotates its one retained generation at STARTUP; the guarantee that holds is an accident of the failure path

### ISSUE THREAD

**Issue by DanBeckDev** [DanBeckDev/a11y-witness#13]
## What it is

`train-screenreader-model.py` rotates the one retained model generation at STARTUP — `rmtree(previous)` then `move(output, previous)` — before the train that might justify it. So a train that crashes has already spent the retained generation before doing any work.

**It cannot currently destroy a release-eligible model, and the reason is an accident rather than a design.** A crashed train writes NO model, so `args.output` is absent or partial afterwards, so it is not release-eligible, so the next run does not reach the rotation branch at all and `.previous` is left alone. Measured 2026-09-06: train #1 rotated a release-eligible v19 aside and died on `torch.stack([])` after the encoder pass; train #2 ran to completion; `.previous` still held the earlier candidate with its acceptance report. Verified by `lab:inventory`, not inferred.

The file's own comment states this and states the consequence: **"If you change WHAT A FAILED TRAIN LEAVES BEHIND, this reasoning expires with it."** That is the hazard — a guarantee that holds through an accident of the failure path, with nothing testing it.

The stronger form is to rotate on SUCCESS: write to a temporary directory and move on success. Deliberately not done at the time, because it restructures the one path that holds release-eligible weights for a hazard measurement says does not currently bite. This issue is that deferral, made findable.

## Region

`packages/lab/scripts/train-screenreader-model.py`

## Acceptance

```
PYTHONDONTWRITEBYTECODE=1 .venv/bin/pytest -p no:cacheprovider <the rotation test>
```
Either (a) rotate on success, with a test that a crashed train leaves `.previous` untouched AND that a partial `output` cannot be mistaken for release-eligible; or (b) leave the startup rotation and add the test that pins the accidental guarantee, so a change to what a failed train leaves behind fails loudly instead of expiring a comment. (b) is legitimate and cheaper — the point is that the guarantee stops being untested, not that the code must move.

## Open-check, run 2026-09-06

```
$ sed -n '236,239p' packages/lab/scripts/train-screenreader-model.py
        previous = args.output.with_name(args.output.name + ".previous")
        if previous.exists():
            shutil.rmtree(previous)
        shutil.move(str(args.output), str(previous))
$ git log --branches='agent/*' --not origin/main --oneline --source -- packages/lab/scripts/train-screenreader-model.py
(empty — unclaimed)
```

Rotation is still at startup, and the deferral comment is still the only thing holding the guarantee.

## Where the full derivation lives

`docs/backlog.md` § "OPEN — five rows I told the CEO were 'on the backlog' and did not write"

## Fleet

No — the change and its test are offline. Do not land it while a train is in flight.

**Comment by DanBeckDev** [comment]
## DISPUTED, and not Ready until it is ruled on

Two readings, both defensible, recorded here rather than settled by whoever writes next:

- **`dispatcher`: closed.** The comment at `train-screenreader-model.py:240-257` states the ordering and the reasoning, which is this row's own second acceptance option ("leave the startup rotation and add the test that pins the accidental guarantee").
- **`worker-config`: open, and the comment is the reason it exists.** *A comment explaining why a fix was deferred is the reason the row exists, not its closure.* Its proposal is to reclassify this as **"Decided — not defects"** — a deliberate, recorded non-defect — rather than queue it as work.

That distinction is worth getting right rather than splitting: "somebody decided not to do this, and here is why" and "somebody has not got to this yet" need opposite handling, and this repo keeps a "Decided — not defects" section precisely so nobody reopens the first kind by mistake.

**Passed to `orchestrator`, not yet ruled on.** Whichever way it goes, the reasoning goes on this issue.

One thing both readings agree on and that survives either verdict: the guarantee currently holds by an accident of what a crashed train leaves behind, and the file says so — *"If you change WHAT A FAILED TRAIN LEAVES BEHIND, this reasoning expires with it."* If it is closed as a non-defect, that sentence is the thing a future change must trip over.

**Comment by DanBeckDev** [comment]
**`ready` removed — I added it to hit the queue floor and it was not pickable.** This row's own thread says *DISPUTED, and not Ready until it is ruled on*, and the ruling has not come back from `orchestrator`.

**That is exactly the defect the ready-label rule was written against**: `ready` is supposed to be a claim about the PRESENT, not a memory, and I made it a claim about a number I wanted to reach. Caught by worker-audit checking the row before taking it, which is the eleventh time today a premise check has beaten a label.

**Comment by DanBeckDev** [comment]
**DECIDED, NOT A DEFECT — ruled by `ceo` after sitting with `orchestrator` all evening. Closing as a decision rather than as a fix.**

**The premise is TRUE and stays true:** the trainer rotates its one retained generation at STARTUP — `rmtree(previous)` then `move(output, previous)` — before the train can fail. Measured 2026-09-06: train #1 rotated the release-eligible model aside and died on `torch.stack([])`. **So the guarantee is weaker than "one generation is kept" reads, and that remains worth knowing.**

**What makes it a decision rather than a defect is WHAT ACTUALLY PROTECTS THE GENERATION**, and it is not the rotation: `training-report.json` is written LAST and ATOMICALLY (`.tmp` then `Path.replace`), so a train that dies anywhere leaves no readable report, `_read_existing_report` returns early, and the next run never reaches the rotation branch. That is why train #2 left `.previous` alone. `train-rotation-safety.test.ts` pins that property with three assertions, each mutation-checked: break the atomic replace, add a second writer, or let a later branch set `releaseEligible` back to True, and one fails.

**Rotate-on-success was considered and deliberately NOT done.** It restructures the one path holding release-eligible weights, for a hazard measurement says does not bite.

**THE CONDITION THAT REOPENS THIS**, stated so the deferral does not have to be re-derived: **if what a FAILED TRAIN LEAVES BEHIND changes.** The reasoning above ends with its own expiry condition, and the test is what remembers it — so a change that makes a crashed train write a readable report, or write its model before its report, turns this straight back into a live defect.

**`worker-config`'s objection stands as the reasoning**, and the reclassification is the point: **a comment recording why a fix was deferred is not a closure and not a defect — it is a decision**, and leaving it labelled `disputed` kept it in a queue nobody could act on.
